Updating collector documentation

GFiuza opened the pull request:
This PR is updating some documentation about the collector with the new collector layout and adding some pages about functionality that did not exist.

Commit:
Updating collector documentation

Commit:
Merge remote-tracking branch 'intelie/main' into main

amadeubarbosa commented on collector/introduction.md:
```diff
@@ -8,6 +8,6 @@ When installed at the rig site, it has many advantages, such as high priority re
 ![](https://lh4.googleusercontent.com/MPecYS1S-WZ17EFA3fEhUms05m8FrdLzYm\_aRKAv0tMTSgML\_5N0xeIeRkePRDS29SLjgoDcE4b853K22V5F-Lq0EPhpL281ientTVuImNFakZJuGdH43LWie6MMGRmpEkNnZNVM)
 
-The collector primarily obtains data through the WITSML protocol; requesting log data from several service companies. Versions 1.3.1.1, 1.4.1.1 and 2.0 are supported. There is also support for the WITS protocol through TCP/IP or serial port, OPC-UA and OPC-DA.
+The collector primarily obtains data through the WITSML protocol; requesting log data from several service companies. Versions 1.3.1.1, 1.4.1.1 and 2.0 are supported. There is also support for the CSV and Raw protocol through TCP/IP or serial port, OPC-UA, OPC-DA and Modbus protocols.
```
Why did we remove the WITS protocol from this phrase? WITS and WITSML are very different. Maybe we could change to something like:

```suggestion
The collector primarily obtains data through the WITSML protocol to request log, trajectory and other object data from service companies. WITSML versions 1.3.1.1, 1.4.1.1 and 2.0 are supported. 

Industry-standard protocols like OPC-UA, OPC-DA and Modbus are also supported. As a fallback for legacy system integrations, LiveRig Collector also can obtain data in WITS (level 0), CSV and Raw data through TCP/IP or serial ports.
```

GFiuza replied:
I was trying to rephrase it so that it doesn't look like WITS support serial port connection. But it looks like I accidentally erased the WITS protocol from the introduction. Anyway, the way you phrased it is way better, I will apply your suggestion. Thanks, Amadeu!

GFiuza replied:
Fixed in the commit 2ef49f35b175a80f91948b0480dabe96ad78bf21

amadeubarbosa commented on SUMMARY.md:
```diff
@@ -164,6 +165,7 @@
   * [sources.xml](collector/configuration/sources.xml.md)
   * [Tags in LiveRig Collector](collector/configuration/tags-in-liverig-collector.md)
   * [Certificate-based authentication for WITSML HTTPS sources](collector/configuration/certificate-based-authentication-for-witsml-https-sources.md)
+  * [store.json](collector/configuration/store.json.md)
```
Missing modbus.json.md under the configuration in SUMMARY.md listing. 

Sorry folks, I forget to explain the SUMMARY.md is important to keep consistent to the new tree of .md, actually when editing by gitbok UI , it ensures the consistency but here at github integration we need to be very careful to those links.

Bold rule:
- if any page is removed, please remove from summary
- if any page is created, please add to the summary

GFiuza replied:
Fixed in the commit 18ed7bf0cbbe79b93720c7873bc3a625920ed593

amadeubarbosa commented on SUMMARY.md:
```diff
@@ -147,8 +147,9 @@
-  * [Requests](collector/remote-control/requests.md)
+  * [Object Explorer](collector/remote-control/object-explorer.md)
   * [WITSML Backlog Sync](collector/remote-control/backlog-sync.md)
+  * [Properties](collector/remote-control/properties.md)
```
Missing `witsml-requests.md`, `opc-requests.md` at this point in the SUMMARY as explained below (in another comment).

GFiuza replied:
Fixed in the commit 658fb61c6ec0b99b0e364406743bb74a265617ab

Commit:
Fixing missing entries in the summary.

Commit:
Readding WITS protocol in the introductory text.

Commit:
Fixing missing entries in the summary

Commit:
Removing duplicated entry in the summary

amadeubarbosa commented on SUMMARY.md:
```diff
@@ -147,8 +147,11 @@
     * [getFromStore](collector/remote-control/apis/getfromstore.md)
   * [Sources](collector/remote-control/sources.md)
-  * [Requests](collector/remote-control/requests.md)
+  * [Witsml Requests](collector/remote-control/witsml-requests.md)
```
```suggestion
  * [WITSML Requests](collector/remote-control/witsml-requests.md)
```

amadeubarbosa commented on SUMMARY.md:
```diff
@@ -147,8 +147,11 @@
   * [Sources](collector/remote-control/sources.md)
-  * [Requests](collector/remote-control/requests.md)
+  * [Witsml Requests](collector/remote-control/witsml-requests.md)
+  * [Opc Requests](collector/remote-control/opc-requests.md)
```
```suggestion
  * [OPC Requests](collector/remote-control/opc-requests.md)
```

amadeubarbosa commented on SUMMARY.md:
```diff
@@ -147,8 +147,11 @@
-  * [Requests](collector/remote-control/requests.md)
+  * [Witsml Requests](collector/remote-control/witsml-requests.md)
+  * [Opc Requests](collector/remote-control/opc-requests.md)
+  * [Object Explorer](collector/remote-control/object-explorer.md)
```
```suggestion
  * [WITSML Object Explorer](collector/remote-control/object-explorer.md)
```

amadeubarbosa replied:
But this will demand the rename of the `remote-control/object-explorer.md` to `remote-control/witsml-object-explorer.md` to keep the gitbook consistency

amadeubarbosa commented on collector/remote-control/witsml-requests.md:
```diff
@@ -0,0 +1,25 @@
+{% hint style="info" %}
+This feature is available on plugin liverig version 4.7.0 or above
```
```suggestion
This feature is available on plugin-liverig version 4.7.0 or above
```

amadeubarbosa commented on collector/remote-control/object-explorer.md:
```diff
@@ -1,6 +1,6 @@
 # Requests
 
-When source has mode client, there is a tab called List of requests on source modal. In this feature, the witsml tree, showing available objects, can be requested from remote control.
+When source has mode client, there is a tab called `Object Explorer` on source modal. In this feature, the witsml tree, showing available objects, can be requested from remote control.
```
```suggestion
Once a WITSML source is configured in client mode, a tab called `Object Explorer` will display at the source details. This feature lists the WITSML objects tree and accelerate the user to configure the requests to be made by the remote control.

Alternatively, the user can configure WITSML requests in a more advanced mode with wildcards support against object ID and names at the `Requests` tab. See [WITSML Requests](./witsml-requests.md) for more details.
```

amadeubarbosa commented on collector/monitoring.md:
```diff
@@ -4,17 +4,21 @@ On Intelie Live `Collectors` tab, it is possible to get an overview of all the a
+![Configuration Tab available for WITSML, WITS, CSV, OPC-DA, OPC-UA, Modbus and Raw sources](<../.gitbook/assets/collector-witsml-source.png>)
 
-![Modal for sources with mode client (Available only for WITSML) - Configurations](<../.gitbook/assets/image (19).png>)
+![Request Tab for Witsml Sources](<../.gitbook/assets/collector-witsml-requests.png>)
```
```suggestion
![Request Tab for WITSML Sources](<../.gitbook/assets/collector-witsml-requests.png>)
```

amadeubarbosa commented on collector/remote-control/witsml-requests.md:
```diff
@@ -0,0 +1,25 @@
+
+![WITSML Requests page showing edited requests in the sidebar with a yellow dot](<../../.gitbook/assets/collector-remote-control-witsml-duplicated-requests-modal.png>)
+
+It's also possible to delete witsml requests when needed. To do so, select the request that should be deleted, and in the upper right corner a `DELETE REQUEST` button should be visible. After clicking it, a confirmation modal will appear asking for confirmation. After clicking the `Delete` button on the confirmation modal, the request will be imediatelly deleted. There is no need to click on the  ...
```
```suggestion
It's also possible to delete WITSML requests when needed. To do so, select the request that should be deleted, and in the upper right corner a `DELETE REQUEST` button should be visible. After clicking it, a confirmation modal will appear asking for confirmation. After clicking the `Delete` button on the confirmation modal, the request will be imediatelly deleted. There is no need to click on the save after the confirmation:
```

amadeubarbosa commented on collector/remote-control/witsml-requests.md:
```diff
@@ -0,0 +1,25 @@
+
+![WITSML Request confirmation modal when deleting a request](<../../.gitbook/assets/collector-remote-control-witsml-delete-modal.png>)
+
+After clicking the "Save" button or deleting a request, some time is needed before the collector apply the modifications.
```
```suggestion
After clicking the `Save` button or deleting a request, some time is needed before the collector apply the modifications in the remote site.
```

amadeubarbosa committed:
Apply suggestions from code review

Review WITSML spelling (to uppercase) and some improvements on the requests documentation.

amadeubarbosa commented on collector/remote-control/object-explorer.md:
```diff
@@ -1,6 +1,8 @@
 # Requests
```
```suggestion
# Object Explorer
```

amadeubarbosa committed:
Rename the heading of collector/remote-control/object-explorer.md

amadeubarbosa committed:
Update and rename object-explorer.md to witsml-object-explorer.md

amadeubarbosa commented on SUMMARY.md:
```diff
@@ -147,8 +147,11 @@
-  * [Requests](collector/remote-control/requests.md)
+  * [WITSML Requests](collector/remote-control/witsml-requests.md)
+  * [OPC Requests](collector/remote-control/opc-requests.md)
+  * [WITSML Object Explorer](collector/remote-control/object-explorer.md)
```
```suggestion
  * [WITSML Object Explorer](collector/remote-control/witsml-object-explorer.md)
```

amadeubarbosa committed:
Fix SUMMARY.md after the rename of object explorer

amadeubarbosa commented on collector/remote-control/README.md:
```diff
@@ -1,7 +1,7 @@
 
 {% hint style="info" %}
-In order to use this functionality, set **`enable_remote_control`** as **`true`**in the [liverig.properties](../configuration/liverig.properties.md) file.enable\_remote\_control\_backlog\_sync
+In order to use this functionality, set **`enable_remote_control`** as **`true`**in the [liverig.properties](../configuration/liverig.properties.md) file.
```
```suggestion
In order to use this functionality, set **`enable_remote_control`** as **`true`** in the [liverig.properties](../configuration/liverig.properties.md) file.
```

amadeubarbosa committed:
Fix small typo in markdown collector/remote-control/README.md

amadeubarbosa commented on collector/remote-control/sources.md:
```diff
@@ -16,10 +16,10 @@ On the Access section, some protocols (example: WITSML and OPC) support the **Te
 On the source page, the **configurations** tab will allow you to update the configured event type and store access information, saving those changes may take a while to update as well.
 
-Dependent upon the protocol type, additional tabs can be displayed. For example, sources for WITSML allow the configuration of [requests](https://drilling.intelie.com/collector/remote-control/requests) and [backlog sync](https://drilling.intelie.com/collector/remote-control/backlog-sync).&#x20;
+Dependent upon the protocol type, additional tabs can be displayed. For example, sources for WITSML allow the configuration of [requests](https://drilling.intelie.com/collector/remote-control/requests), [backlog sync](https://drilling.intelie.com/collector/remote-control/backlog-sync) and [object explorer](https://drilling.intelie.com/collector/remote-control/object-explorer).&#x20;
```
```suggestion
Dependent upon the protocol type, additional tabs can be displayed. For example, sources for WITSML allow the configuration of [requests](https://drilling.intelie.com/collector/remote-control/requests), [backlog sync](https://drilling.intelie.com/collector/remote-control/backlog-sync) and [object explorer](https://drilling.intelie.com/collector/remote-control/witsml-object-explorer).
```

amadeubarbosa committed:
Update link for object explorer at collector/remote-control/sources.md

amadeubarbosa committed:
Reorganizing collector -> sources as a subsection

amadeubarbosa committed:
Rename backlog sync to witsml

GFiuza committed:
Merge pull request #1 from amadeubarbosa/review-over-fiuza-changes-pr-5

Review over fiuza changes pr 5

amadeubarbosa committed:
Fixes link references at README on sources section

amadeubarbosa committed:
Fixes link references at opc-requests on sources section

amadeubarbosa committed:
Fixes WITSML backlog sync link reference

amadeubarbosa committed:
Fixes requests and object explorer link references

amadeubarbosa committed:
Fixes image link references for WITSML requests

amadeubarbosa committed:
Fixes link references at WITSML backlog sync

amadeubarbosa committed:
Fixes image references at WITSML Object Explorer
